brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java

```diff
@@ -459,7 +459,7 @@ public boolean buildResourcesFull(File appDir, UsesFramework usesFramework)
             }
             File apkDir = new File(appDir, APK_DIRNAME);
             if (apkOptions.forceBuildAll || isModified(newFiles(APP_RESOURCES_FILENAMES, appDir),
-                    newFiles(APK_RESOURCES_FILENAMES, apkDir))) {
+                    newFiles(APK_RESOURCES_FILENAMES, apkDir)) || apkOptions.isAapt2()) {
```

Comment:
This is because we aren't storing the "intermediate" file anywhere. Lets dump this intermediate file to our "build" directory, let `aapt1` ignore it and `aapt2` if smart enough can detect if its modified or not. So we can skip this `compile` stage if nothing has changed.